src/porepy/grids/md_grid.py

```diff
@@ -733,7 +819,7 @@ def dim_max(self) -> int:
         """
         # Access protected attribute instead of subdomains() to avoid sorting, which can
         # lead to a recursion error.
-        return np.max([sd.dim for sd in self._subdomain_data.keys()])
+        return max([sd.dim for sd in self._subdomain_data.keys()], default=0)
```

Comment:
Any particular reason to have a default, and set this default value? I can imagine getting a 0 here will give trouble, e.g., for calculations of grid refinement levels.